addons/mail/models/discuss/discuss_channel.py

```diff
@@ -473,14 +473,33 @@ def _action_unfollow(self, partner=None, guest=None, post_leave_message=True):
             ],
         )
 
+    def _get_add_members_allowed_create_params(self):
+        """Get the allowed parameters to pass to create member through
+        add_members"""
+        return []
+
     def add_members(
-        self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False, post_joined_message=True
+        self,
+        partner_ids=None,
+        guest_ids=None,
+        invite_to_rtc_call=False,
+        create_member_kwargs=None,
+        post_joined_message=True,
     ):
         """ Adds the given partner_ids and guest_ids as member of self channels. """
         current_partner, current_guest = self.env["res.partner"]._get_current_persona()
         partners = self.env['res.partner'].browse(partner_ids or []).exists()
         guests = self.env['mail.guest'].browse(guest_ids or []).exists()
         all_new_members = self.env["discuss.channel.member"]
+        create_member_params = (
+            {
+                k: v
+                for k, v in create_member_kwargs.items()
+                if k in self._get_add_members_allowed_create_params()
+            }
+            if create_member_kwargs
+            else {}
+        )
```

Comment:
Probably not worth the effort.

We might want to make `add_members` private, and have a route for calls from the UI with limited parameters.